spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/LocalOptimizer.scala

```diff
@@ -152,8 +152,19 @@ class LocalOptimizer[T: ClassTag] (
           })
       )
       val loss = lossSum / parallelism
-      grad.div(ev.fromType(parallelism))
+      var scale = ev.fromType(parallelism)
+      if (gradientClippingParams.enableL2NormClipping) {
```

Comment:
If the model (e.g., AlexNet or VGG) has large parameter sizes, the overhead can be large? Shall we do this using multi-threading?

Reply:
Updated to use multi-thread to compute the square sum